SimTKcommon/tests/TestStateXml.cpp

```diff
@@ -25,6 +25,46 @@
 
 using namespace SimTK;
 
+/* This function converts the provided instance of `T` to Xml, deserializes
+it from Xml, ensures the deserialized instance is equal to the original
+instance, reserializes the deserialized instance back to Xml and ensures
+the original desrialization and the reserialization are equal.
+
+One might set `checkInMemoryEquality` to false if `value` contains NaNs. */
+template <typename T>
+void testXmlRoundtrip(const T& value, bool checkInMemoryEquality = true,
```

Comment:
I like this function -- that's a very good (stringent) test for (de)serialization.